src/Microsoft.Extensions.Configuration.UserSecrets/ConfigurationExtensions.cs

```diff
@@ -76,15 +24,15 @@ public static IConfigurationBuilder AddUserSecrets(this IConfigurationBuilder co
                 throw new ArgumentNullException(nameof(configuration));
             }
 
-            var projectPath = configuration.GetProjectPath();
-            if (string.IsNullOrEmpty(projectPath))
+            var fileProvider = configuration.GetFileProvider();
+            if (fileProvider == null)
             {
                 throw new InvalidOperationException(Resources.FormatError_MissingBasePath(
-                        projectPath,
-                        typeof(IConfigurationBuilder).Name,
-                        ProjectPathKey));
+                        nameof(IConfigurationBuilder),
+                        nameof(IFileProvider)));
             }
-            return AddSecretsFile(configuration, PathHelper.GetSecretsPath(projectPath));
+
+            return AddSecretsFile(configuration, PathHelper.GetSecretsPath(fileProvider.GetFileInfo("/").PhysicalPath));
```

Comment:
So there's one scenario that no-opping is a problem. When in the dev environment, you publish your app, and you want to test the published app, so you don't have a project.json. If we throw an exception with a good explanation, it would be better IMO. Otherwise folks may have a more obscure error later.